x-pack/plugins/infra/public/components/logging/log_source_error_page.tsx

```diff
@@ -0,0 +1,141 @@
+          <>
+            <p>
+              <FormattedMessage
+                id="xpack.infra.logSourceErrorPage.failedToLoadSourceMessage"
+                defaultMessage="Errors occurred while attempting to load the configuration. Try again or change the configuration to fix the problem."
+              />
+            </p>
+            {errors.map((error) => (
+              <React.Fragment key={error.name}>
+                <LogSourceErrorMessage error={error} />
+                <EuiSpacer />
+              </React.Fragment>
+            ))}
+          </>
+        }
+        actions={[
+          <EuiButton onClick={onRetry} iconType="refresh" fill>
+            <FormattedMessage
+              id="xpack.infra.logSourceErrorPage.tryAgainButtonLabel"
+              defaultMessage="Try again"
+            />
+          </EuiButton>,
+          <EuiButtonEmpty iconType="gear" {...settingsLinkProps}>
+            <FormattedMessage
+              id="xpack.infra.logSourceErrorPage.navigateToSettingsButtonLabel"
+              defaultMessage="Change configuration"
+            />
+          </EuiButtonEmpty>,
+        ]}
+      />
+    </EuiPageTemplate>
+  );
+};
+
+const LogSourceErrorMessage: React.FC<{ error: Error }> = ({ error }) => {
+  if (error instanceof ResolveLogSourceConfigurationError) {
+    return (
+      <LogSourceErrorCallout
+        title={
+          <FormattedMessage
+            id="xpack.infra.logSourceErrorPage.resolveLogSourceConfigurationErrorTitle"
+            defaultMessage="Failed to resolve the log source configuration"
+          />
+        }
+      >
+        {error.cause instanceof SavedObjectNotFound ? (
+          // the SavedObjectNotFound error message contains broken markup
+          <FormattedMessage
+            id="xpack.infra.logSourceErrorPage.savedObjectNotFoundErrorMessage"
+            defaultMessage="Failed to locate that {savedObjectType}: {savedObjectId}"
+            values={{
+              savedObjectType: error.cause.savedObjectType,
+              savedObjectId: error.cause.savedObjectId,
+            }}
+          />
+        ) : (
+          `${error.cause?.message ?? error.message}`
+        )}
+      </LogSourceErrorCallout>
+    );
+  } else if (error instanceof FetchLogSourceConfigurationError) {
+    return (
+      <LogSourceErrorCallout
+        title={
+          <FormattedMessage
+            id="xpack.infra.logSourceErrorPage.fetchLogSourceConfigurationErrorTitle"
+            defaultMessage="Failed to load the log source configuration"
+          />
+        }
+      >
+        {`${error.cause?.message ?? error.message}`}
+      </LogSourceErrorCallout>
+    );
+  } else if (error instanceof FetchLogSourceStatusError) {
+    return (
+      <LogSourceErrorCallout
+        title={
+          <FormattedMessage
+            id="xpack.infra.logSourceErrorPage.fetchLogSourceStatusErrorTitle"
+            defaultMessage="Failed to determine the status of the log source"
```

Comment:
The other errors use `log source configuration` should we add `configuration` onto this one?

Reply:
Yes, I was unsure about this too. I figured this is about the status of the log source and not really about the status of the configuration of the log source. I don't have a strong opinion, but I'm eager to merge this before Jenkins changes its mind. :grimacing: